File: Examples/CSharp/AsposePDF/Stamps-Watermarks/PageNumberStamps.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Text;

namespace CSharp.AsposePdf.StampsWatermarks
{
    public class PageNumberStamps
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_StampsWatermarks();

            //open document
            Document pdfDocument = new Document(dataDir+ "PageNumberStamp.pdf");

            //create page number stamp
            PageNumberStamp pageNumberStamp = new PageNumberStamp();
            //whether the stamp is background
            pageNumberStamp.Background = false;
            pageNumberStamp.Format = "Page # of " + pdfDocument.Pages.Count;
            pageNumberStamp.BottomMargin = 10;
            pageNumberStamp.HorizontalAlignment = HorizontalAlignment.Center;
            pageNumberStamp.StartingNumber = 1;
            //set text properties
            pageNumberStamp.TextState.Font = FontRepository.FindFont("Arial");
            pageNumberStamp.TextState.FontSize = 14.0F;
            pageNumberStamp.TextState.FontStyle = FontStyles.Bold;
            pageNumberStamp.TextState.FontStyle = FontStyles.Italic;
            pageNumberStamp.TextState.ForegroundColor = Color.Aqua;

            //add stamp to particular page
            pdfDocument.Pages[1].AddStamp(pageNumberStamp);

            //save output document
            pdfDocument.Save(dataDir+ "PageNumberStamp_out.pdf");
 
            
            
        }
    }
}

File: Examples/VisualBasic/AsposePDF/Stamps-Watermarks/PageNumberStamps.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf
Imports Aspose.Pdf.Text

Namespace AsposePDF.StampsWatermarks
    Public Class PageNumberStamps
        Public Shared Sub Run()
            ' ExStart:PageNumberStamps
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_StampsWatermarks()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("PageNumberStamp.pdf"))

            ' Create page number stamp
            Dim pageNumberStamp As New PageNumberStamp()
            ' Whether the stamp is background
            pageNumberStamp.Background = False
            pageNumberStamp.Format = "Page # of " + pdfDocument.Pages.Count.ToString()
            pageNumberStamp.BottomMargin = 10
            pageNumberStamp.HorizontalAlignment = HorizontalAlignment.Center
            pageNumberStamp.StartingNumber = 1
            ' Set text properties
            pageNumberStamp.TextState.Font = FontRepository.FindFont("Arial")
            pageNumberStamp.TextState.FontSize = 14.0F
            pageNumberStamp.TextState.FontStyle = FontStyles.Bold
            pageNumberStamp.TextState.FontStyle = FontStyles.Italic
            pageNumberStamp.TextState.ForegroundColor = Color.Aqua

            ' Add stamp to particular page
            pdfDocument.Pages(1).AddStamp(pageNumberStamp)

            dataDir = dataDir & Convert.ToString("PageNumberStamp_out_.pdf")
            ' Save output document
            pdfDocument.Save(dataDir)
            ' ExEnd:PageNumberStamps
            Console.WriteLine(Convert.ToString(vbLf & "Page number stamp added successfully." & vbLf & "File saved at ") & dataDir)

        End Sub
    End Class
End Namespace
